recipes-extended/xen/blktap3/use-existing-control-file.patch

```diff
@@ -0,0 +1,59 @@
+################################################################################
+SHORT DESCRIPTION: 
+################################################################################
+Use existing(created by blktap kernel module) control file.
+
+################################################################################
+LONG DESCRIPTION: 
+################################################################################
+Instead of creating a new control file, use the one that is already created by
+blktap kernel module.
+
+################################################################################
+CHANGELOG 
+################################################################################
+Author(s):
+Mahantesh Salimath <salimathm@ainfosec.com>
+
+################################################################################
+REMOVAL 
+################################################################################
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+Yes, could be upstreamed.
+
+################################################################################
+INTERNAL DEPENDENCIES 
+################################################################################
+
+################################################################################
+PATCHES 
+################################################################################
+diff --git a/include/blktap.h b/include/blktap.h
+index a78d7da..1fd0737 100644
+--- a/include/blktap.h
++++ b/include/blktap.h
+@@ -36,7 +36,7 @@
+ #define BLKTAP2_CONTROL_DIR            "/var/run/blktap-control"
+ #define BLKTAP2_CONTROL_SOCKET         "ctl"
+ #define BLKTAP2_DIRECTORY              "/dev/xen/blktap-2"
+-#define BLKTAP2_CONTROL_DEVICE         BLKTAP2_DIRECTORY"/control"
++#define BLKTAP2_CONTROL_DEVICE         "/dev/blktap-control"
```

Comment:
Can we just use udev to create a symlink instead of patching the file?

Reply:
I tried creating symlink using udev rule. When I run udevadm test the logs show that rule should run fine but when I reboot the machine the udev rule seems to not have been triggered. This may be because it is not an actual device, but not sure. Also selinux policy needs to be updated in this case. 
However, I think this not much of concern, this hunk can be dropped without any issue.
Let me know what you think, keep it or drop it ?

Reply:
Hmm, looks like the code creates BLKTAP2_CONTROL_DEVICE if it doesn't exist.  I like your idea of using /dev/blktap-control, but I would just drop the hunk to minimize the diff.

Reply:
Ack, will do.